src/compiler-src.md

```diff
@@ -1,137 +1,191 @@
 # High-level overview of the compiler source
 
-## Crate structure
-
-The main Rust repository consists of a `src` directory, under which
-there live many crates. These crates contain the sources for the
-standard library and the compiler.  This document, of course, focuses
-on the latter.
-
-Rustc consists of a number of crates, including `rustc_ast`,
-`rustc`, `rustc_target`, `rustc_codegen`, `rustc_driver`, and
-many more. The source for each crate can be found in a directory
-like `src/libXXX`, where `XXX` is the crate name.
-
-(N.B. The names and divisions of these crates are not set in
-stone and may change over time. For the time being, we tend towards a
-finer-grained division to help with compilation time, though as incremental
-compilation improves, that may change.)
-
-The dependency structure of these crates is roughly a diamond:
-
-```text
-                  rustc_driver
-                /      |       \
-              /        |         \
-            /          |           \
-          /            v             \
-rustc_codegen  rustc_borrowck   ...  rustc_metadata
-          \            |            /
-            \          |          /
-              \        |        /
-                \      v      /
-                  rustc_middle
-                       |
-                       v
-                   rustc_ast
-                    /    \
-                  /       \
-           rustc_span  rustc_builtin_macros
-```
```

Comment:
This is a really neat diagram, could we keep it (with crate names updated)?

Reply:
In the last month or so I believe that there was discussion about mdbook support for diagrams.  Has this been implemented? We could work in a number of diagrams across the guide.

Reply:
I think it's more likely for us to use something like mermaid. I believe the chalk book does that.

The hard part is that the crate graph is rather complicated and no longer looks like the original graph.

So I would like to try to merge this and add useful diagrams later.

Reply:
> The hard part is that the crate graph is rather complicated and no longer looks like the original graph.

I see, that's unfortunate.